From a8e905fb3fd0d26f724646275b72a7363b2f03d8 Mon Sep 17 00:00:00 2001
From: Max Fierke <max@maxfierke.com>
Date: Wed, 1 Jun 2022 00:17:47 -0500
Subject: [PATCH 015/117] dt-bindings: display: Add bindings for ClockworkPi
 CWD686

The CWD686 is a 6.86" IPS LCD panel used as the primary
display in the ClockworkPi DevTerm portable (all cores)

Signed-off-by: Max Fierke <max@maxfierke.com>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 .../display/panel/clockwork,cwd686.yaml       | 62 +++++++++++++++++++
 1 file changed, 62 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/panel/clockwork,cwd686.yaml

--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/clockwork,cwd686.yaml
@@ -0,0 +1,62 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/clockwork,cwd686.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Clockwork CWD686 6.86" IPS LCD panel
+
+maintainers:
+  - Max Fierke <max@maxfierke.com>
+
+description: |
+  The Clockwork CWD686 is a 6.86" ICNL9707-based IPS LCD panel used within the
+  Clockwork DevTerm series of portable devices. The panel has a 480x1280
+  resolution and uses 24 bit RGB per pixel.
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    const: clockwork,cwd686
+
+  reg:
+    description: DSI virtual channel used by that screen
+    maxItems: 1
+
+  reset-gpios: true
+  rotation: true
+  backlight: true
+  iovcc-supply: true
+  vci-supply: true
+
+required:
+  - compatible
+  - reg
+  - backlight
+  - reset-gpios
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    backlight: backlight {
+        compatible = "gpio-backlight";
+        gpios = <&gpio4 30 GPIO_ACTIVE_HIGH>;
+    };
+
+    dsi {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        panel@0 {
+            compatible = "clockwork,cwd686";
+            reg = <0>;
+            backlight = <&backlight>;
+            reset-gpios = <&gpio2 28 GPIO_ACTIVE_HIGH>;
+            rotation = <90>;
+        };
+    };
